client/extensions/woocommerce/app/products/product-form.js

```diff
@@ -11,15 +11,20 @@ import classNames from 'classnames';
 /**
  * Internal dependencies
  */
+import { getProductEdits } from 'woocommerce/state/ui/products/selectors';
+import { getProductVariationsWithLocalEdits } from 'woocommerce/state/ui/products/variations/selectors';
+import { getProductCategoriesWithLocalEdits } from 'woocommerce/state/ui/product-categories/selectors';
 import ProductFormAdditionalDetailsCard from './product-form-additional-details-card';
 import ProductFormCategoriesCard from './product-form-categories-card';
 import ProductFormDetailsCard from './product-form-details-card';
 import ProductFormSimpleCard from './product-form-simple-card';
 import ProductFormVariationsCard from './product-form-variations-card';
+import { withWooCommerceSite } from 'woocommerce/state/wc-api';
 
-export default class ProductForm extends Component {
+class ProductForm extends Component {
 	static propTypes = {
 		className: PropTypes.string,
+		wcApiSite: PropTypes.any,
```

Comment:
so is this a site object or a site ID? since it is used in `getProductEdits` I assume it is a an ID/number - so shouldn't the proptype be that? Or maybe this should be `wcApiSiteId` to make it clear.

Reply:
I originally planned on it being an object that could be defined with properties that identify the site within whatever run context is appropriate. For wpcom, this would have been `{ siteId: <number> }` as that is what is used to identify sites here. For use outside of wpcom, in say, wp-admin, this might be `{ baseUrl: 'https://mysubdomain.mydomain.com/' }` to be multisite compatible.

However, I don't know that an object is really necessary here, and it interferes with comparing as new objects are created to represent the same site. So I simplified it to just a number to represent the site on wpcom. However, for a wp-admin setup with multisite, it would still need to be a string url.

For that reason, I'm reluctant to add "Id" to the name of this property.

Reply:
Why not take two different props, a `siteId` or a `baseUrl`? This way you can accept the correct propType & make verification a little easier. Technically both would be optional, but you could always fire off a `debug` message or something (or throw an error, even), if neither are set. If both are set, I'd say preference should be given to `baseUrl`, since a dev specifying a URL probably expects that to be used.

Reply:
Well, I'm trying to avoid properties that are implementation specific here, but maybe that's not so valuable.

I think maybe instead of making the component "all-purpose" here, maybe it should be the point of abstraction itself, meaning the wpcom implementation would require a `siteId` and the wp-admin implementation would require a `baseUrl` prop. Thoughts on that?

Reply:
I can see value in the dual prop approach @ryelle outlined above, but also agree it could be a premature addition since this isn't going to be used immediately in wp-admin. Given the nature of this code, and where we are likely heading with it, I think this is still a good thing to 🤔 about and discuss here.

Back here in Calypso land - siteID is such a standard everywhere that is why it still feels like this should be `siteId`

Reply:
> I think maybe instead of making the component "all-purpose" here, maybe it should be the point of abstraction itself

What would this mean for a developer interacting with this component?

> I can see value in the dual prop approach @ryelle outlined above, but also agree it could be a premature addition since this isn't going to be used immediately in wp-admin. 

I think this approach fits into that idea perfectly though, you wouldn't need to add the `baseUrl` prop until we're ready to use this in wp-admin

Reply:
I think we need to be very careful here, if we care about the reusability of these components. If we create different properties for different environments, then we can no longer use the same code in both wpcom and wp-admin. In the case of a generic site prop, the same application code can be used in either place with zero modification. But if we split these props, then `ProductCreate`, `ProductUpdate` and `ProductForm` have to either be different between wpcom and wp-admin, or they have to pass down a prop that is wholly unused in its environment. (I understand these components may not be the actual unified UI anyway, but I think the concern still stands even if they are blocks or something else).